src/pages/SubmissionConfig.js

```diff
@@ -370,6 +372,151 @@ const renderGrid = props => {
+          cellRenderer: () => {
+            return h(TextCell, { style: { fontWeight: 500 } }, [method ? method.name : 'Loading...'])
+          }
+        },
+        {
+          size: { basis: 360, grow: 0 },
+          field: 'workflowVariable',
+          headerRenderer: () => h(Sortable, { sort: inputTableSort, field: 'workflowVariable', onSort: setInputTableSort }, [h(HeaderCell, ['Variable'])]),
+          cellRenderer: ({ rowIndex }) => {
+            return h(TextCell, {}, [configuredInputDefinition[rowIndex].input_name])
+          }
+        },
+        {
+          size: { basis: 160, grow: 0 },
+          headerRenderer: () => h(HeaderCell, ['Type']),
+          cellRenderer: ({ rowIndex }) => {
+            return h(TextCell, {}, [parseInputType(configuredInputDefinition[rowIndex].input_type)]) // TODO: this needs to be more flexible
+          }
+        },
+        {
+          size: { basis: 350, grow: 0 },
+          headerRenderer: () => h(HeaderCell, ['Input sources']),
+          cellRenderer: ({ rowIndex }) => {
+            return h(Select, {
+              isDisabled: false,
+              'aria-label': 'Select an Option',
+              isClearable: false,
+              value: _.get(_.get(`${rowIndex}.source.type`, configuredInputDefinition), inputSourceLabels) || null,
+              onChange: ({ value }) => {
+                const newType = _.get(value, inputSourceTypes)
+                const param = newType === 'record_lookup' ? 'record_attribute' : 'parameter_value'
+                const newSource = {
+                  type: newType,
+                  [param]: _.get(`${rowIndex}.${param}`, cachedInputSources)
+                }
+                const newConfig = _.set(`${rowIndex}.source`, newSource, configuredInputDefinition)
+                setConfiguredInputDefinition(newConfig)
+              },
+              placeholder: 'Select Source',
+              options: _.values(inputSourceLabels),
+              // ** https://stackoverflow.com/questions/55830799/how-to-change-zindex-in-react-select-drowpdown
+              styles: { container: old => ({ ...old, display: 'inline-block' }), menuPortal: base => ({ ...base, zIndex: 9999 }) },
+              menuPortalTarget: document.body,
+              menuPlacement: 'top'
+            })
+          }
+        },
+        {
+          headerRenderer: () => h(Fragment, [
+            h(HeaderCell, ['Attribute'])
+          ]),
+          cellRenderer: ({ rowIndex }) => {
+            const source = _.get(`${rowIndex}.source`, configuredInputDefinition)
+            return Utils.switchCase(source.type || 'default',
+              ['record_lookup', () => recordLookupSelect(rowIndex)],
+              ['literal', () => parameterValueSelect(rowIndex)],
+              ['default', () => h(TextCell, {}, ['Use value from Workflow'])]
```

Comment:
when I saw the mockup, I thought the "use default" option should figure out what the default variable would be, and then display it here - but that would require us to actually parse the WDL, which seemed out-of-scope.

I want to confirm that this should literally say "Use value from Workflow"...

Reply:
We opted to say "Use value..." (as opposed to "Default value...") since the other options are all verbs/calls to action as well, right?

Reply:
> Yeah I think the "value from WDL" is going to need more thinking about (in particular, at workflow parsing time). For now I think we can get the same effect by prepopulating the WDL values into the template run set in the database, and calling it a suggested "literal" value.

this sounds like a thing to deal with in another ticket -- assuming I'm interpreting that correctly, I'll make updates based on your following comment (i.e. change the description, only allow this for optional inputs)